### PR TITLE
Remove the makefile based build system

### PR DESCRIPTION
**Pull request by esden**
## Detailed description

As we are already using meson for a while with pretty good success and the gnu makefile based system is starting to require maintenance effort. It is time to retire the makefiles.

If it turns out that we need to bring them back for some reason it should still be relatively easy to do.

Still this is an end of an era. :)

As a side-effect this PR also deletes the libopencm3 submodule as it is not needed any more.

## Your checklist for this pull request

* [x] I've read the [Code of Conduct](https://github.com/blackmagic-debug/blackmagic/blob/main/CODE_OF_CONDUCT.md)
* [x] I've read the [guidelines for contributing](https://github.com/blackmagic-debug/blackmagic/blob/main/CONTRIBUTING.md) to this repository
* [x] It builds for hardware native (see [Building the firmware](https://github.com/blackmagic-debug/blackmagic?tab=readme-ov-file#building-black-magic-debug-firmware))
* [x] It builds as BMDA (see [Building the BMDA](https://github.com/blackmagic-debug/blackmagic?tab=readme-ov-file#building-black-magic-debug-app))
* [x] I've tested it to the best of my ability
* [x] My commit messages provide a useful short description of what the commits do

### CODE REVIEW

File: src/target/flashstub/Makefile
Comment (dragonmux) [inline src/target/flashstub/Makefile:1]: This one Makefile needs to remain for now to allow rebuilding the Flash stubs which is only done on demand (manually). Aki actually had an idea on how to replace this nicely yesterday, which we'll do in the upcoming feature/better-targets-control as that'll be a welcome removal and allow the stubs to automatically be rebuilt as necessary.

**Review by dragonmux — CHANGES_REQUESTED** [review]
There's one Makefile that's being removed which should not be for the moment and isn't called by the outgoing build system (it was always only ever used manually). Beyond that, this is all looking good and while melancholy to see the old system go, it is indeed starting to cause some aggravating problems and becoming a painful maintenance burden.

File: src/target/flashstub/Makefile
Comment (esden) [reply src/target/flashstub/Makefile:1]: Ahh yeah... good point. Sorry, was bit too eager removing the Makefiles. I will bring it back.

**Comment by sidprice** [comment]
(Almost) End of an era! But what an era!

File: src/target/flashstub/Makefile
Comment (esden) [reply src/target/flashstub/Makefile:1]: Ok, I brought back the stub makefile. It should be removed as part of your better-targets-control PR. :)

**Review by dragonmux — APPROVED** [review]
This all LGTM, merging.

**Pull request merged**